File: backend/src/main/java/com/chalkak/backend/post/api/v1/dto/response/PostCalendarResponse.java
package com.chalkak.backend.post.api.v1.dto.response;

import com.chalkak.backend.post.domain.ModerationStatus;
import com.chalkak.backend.post.service.PostCalendarResult;
import io.swagger.v3.oas.annotations.media.Schema;
import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

public record PostCalendarResponse(
        int year,
        int month,
        List<CalendarPostResponse> posts
) {

    public static PostCalendarResponse from(PostCalendarResult result) {
        return new PostCalendarResponse(
                result.year(),
                result.month(),
                result.posts().stream()
                        .map(CalendarPostResponse::from)
                        .toList()
        );
    }

    public record CalendarPostResponse(
            LocalDate topicDate,
            UUID postId,
            String thumbnailImageUrl,
            @Schema(allowableValues = {"APPROVED", "PENDING"})
            ModerationStatus status
    ) {

        private static CalendarPostResponse from(PostCalendarResult.PostSummary post) {
            return new CalendarPostResponse(
                    post.topicDate(),
                    post.postId(),
                    post.thumbnailImageUrl(),
                    post.status()
            );
        }
    }
}

File: client/android/app/src/main/java/com/stonefive/chalkak/data/remote/post/model/PostCalendarResponse.kt
package com.stonefive.chalkak.data.remote.post.model

import kotlinx.serialization.Serializable

@Serializable
data class PostCalendarResponse(
    val year: Int,
    val month: Int,
    val posts: List<PostCalendarItemResponse>,
)

@Serializable
data class PostCalendarItemResponse(
    val topicDate: String,
    val postId: String,
    val thumbnailImageUrl: String,
    val status: String,
)
